### PR TITLE
DR2-1668 Use RATE for DLQ alarms.

### PR DESCRIPTION
**Pull request by MancunianSam**
The issue is that if we get a message in a DLQ where there is already a
message in the DLQ, we don't get another notification.

Repeating messages for alarms that don't change state is a bit
complicated so we'll go with this as a solution now.

The period is 60 seconds so we shouldn't get massively spammed by a huge
number of messages entering the DLQ at once.

### CODE REVIEW

File: sqs/main.tf [hunk 1]
```diff
@@ -64,20 +64,42 @@ resource "aws_sqs_queue" "dlq_with_sse" {
...
-    QueueName = local.sqs_dlq.name
+  evaluation_periods  = 60
+  metric_query {
+    id = "m1"
+    metric {
+      dimensions = {
+        QueueName = "${var.queue_name}-dlq"
+      }
+      metric_name = "ApproximateNumberOfMessagesVisible"
+      period      = 60
```
Comment (techncl) [inline sqs/main.tf:78]: Just to add, should/could these `period`s be added to a variable?

File: sqs/main.tf [hunk 2]
```diff
@@ -64,20 +64,42 @@ resource "aws_sqs_queue" "dlq_with_sse" {
...
+    id = "m2"
+    metric {
+      dimensions = {
+        QueueName = "${var.queue_name}-dlq"
+      }
+      metric_name = "ApproximateNumberOfMessagesVisible"
+      period      = 60
+      stat        = "Maximum"
+      namespace   = "AWS/SQS"
+    }
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply sqs/main.tf:93]: Ah good spot, one should be `ApproximateNumberOfMessagesNotVisible` `ApproximateNumberOfMessagesVisible` is messages that aren't in flight, being processed by a consumer and `ApproximateNumberOfMessagesNotVisible` is the opposite.

A message can be either visible or not so this way we get the total number of messages in the queue.

File: sqs/main.tf (hunk 1 above)
Comment (MancunianSam) [reply sqs/main.tf]: Yeah, can do. I've added a single variable as I can't see why we'd want different evaluation periods for the different metrics. I've defaulted them to 60 as this seems sensible.

**Review by anthonyhashemi — APPROVED** [review]
Looks good 👍

**Pull request merged**